### PR TITLE
playlist-management

### PR DESCRIPTION
**Pull request by ranitasahaa1230**
Added playlist management, create a playlist, delete a playlist, remove the video from playlist

image
image
image

### CODE REVIEW

File: src/components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -9,21 +9,23 @@ import {
...
   const { showToast } = useToast();
   const navigate = useNavigate();
+  const {
+    state: { isAuth },dispatch,
+  } = useAuth();
 
   const logoutHandler = () => {
-    // localStorage.removeItem("login");
-    // localStorage.removeItem("user");
-    // localStorage.removeItem("signup");
```
Comment (jeetbhalani15) [inline src/components/Sidebar/Sidebar.js:24]: remove comments if not needed

File: src/components/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -71,12 +73,16 @@ export const Sidebar = () => {
...
-      <li onClick={logoutHandler}>
-        <MdExitToApp size={25} />
-        <span>Log Out</span>
-      </li>
+      {isAuth && (
+        <>
+          <li onClick={logoutHandler}>
+            <MdExitToApp size={25} />
+            <span>Log Out</span>
+          </li>
```
Comment (rohanmathur91) [inline src/components/Sidebar/Sidebar.js:82]: semantic is not correct, `li` is a list item so it should have a parent, `ul` or `ol`, also in this file you have put 
`<NavLink to="/history" className="navlink">
        <li>
          <MdHistory size={25} />
          <span>History</span>
        </li>
      </NavLink>`

so `NavLink` should be the not be outside of the `li`, the order should be `ul` -> `li` -> `NavLink`

File: src/components/VideoCard/VideoCard.css [hunk 3]
```diff
@@ -50,22 +50,115 @@
...
+
+li {
+  list-style: none;
+}
+
+.display__options {
+  position: absolute;
+  top: 83%;
+  right: -10%;
+  z-index: 999;
```
Comment (rohanmathur91) [inline src/components/VideoCard/VideoCard.css]: why such a high z-index?

File: src/components/VideoCard/VideoCard.js [hunk 4]
```diff
@@ -1,11 +1,67 @@
...
 
 export const VideoCard = ({ video }) => {
   const { _id, title, creator, uploaded, thumbnail, views, time } = video;
   const navigate = useNavigate();
+  const [displayOptions, setDisplayOptions] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const { showToast } = useToast();
+
+  // const [inputText, setInputText] = useState("");
+  // const [list, setList] = useState([]);
```
Comment (rohanmathur91) [inline src/components/VideoCard/VideoCard.js]: remove this if not needed

**Review by jeetbhalani15 — APPROVED** [review]
Playlist Context & reducer are working fine
-  variables are properly Named.
-  Readable Code & rest all looks good.

File: src/pages/Auth/Login.js [hunk 5]
```diff
@@ -28,10 +28,11 @@ export function Login() {
       localStorage.setItem("user", JSON.stringify(user));
       localStorage.setItem("token", encodedToken);
       if (location.state !== null) {
-        navigate(location?.state?.from?.pathname);
+        navigate(location.state?.from?.pathname);
       } else {
         navigate("/");
       }
+      // navigate(location.state?.from?.pathname || "/",{replace:true});
```
Comment (rohanmathur91) [inline src/pages/Auth/Login.js]: remove commented code

File: src/reducers/playListReducer.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        likedVideos: [...action.payload],
+      };
+
+    // case REMOVE_LIKE_VIDEOS:
+    //   return {
+    //     ...playListState,
+    //     likedVideos: playListState.likedVideos.filter(
+    //       ({ _id }) => _id !== action.payload
+    //     ),
+    //   };
```
Comment (rohanmathur91) [inline src/reducers/playListReducer.js:30]: remove commented code

**Review by rohanmathur91 — APPROVED** [review]
- Remove commented code
- Good that you have declared the action types separately as constants
- Overall looks good, suggested a few changes you can look at them

File: src/components/VideoCard/VideoCard.css (hunk 3 above)
Comment (ranitasahaa1230) [reply src/components/VideoCard/VideoCard.css]: removed it.

File: src/components/Sidebar/Sidebar.js (hunk 2 above)
Comment (ranitasahaa1230) [reply src/components/Sidebar/Sidebar.js:82]: will fix it. Thank you

**Pull request merged**